dhis-2/dhis-api/src/test/java/org/hisp/dhis/eventvisualization/SimpleDimensionHandlerTest.java

```diff
@@ -0,0 +1,142 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.eventvisualization;
+
+import static java.util.Arrays.asList;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hisp.dhis.common.DimensionType.PERIOD;
+import static org.hisp.dhis.eventvisualization.Attribute.COLUMN;
+import static org.hisp.dhis.eventvisualization.SimpleDimension.Type.EVENT_DATE;
+import static org.hisp.dhis.eventvisualization.SimpleDimension.Type.INCIDENT_DATE;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import org.hisp.dhis.common.BaseDimensionalObject;
+import org.hisp.dhis.common.DimensionalObject;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit tests for {@link SimpleDimensionHandler}.
+ *
+ * @author maikel arabori
+ */
+class SimpleDimensionHandlerTest
+{
+
+    @Test
+    void testGetDimensionalObject()
+    {
+        // Given
+        final EventVisualization aEventVisualization = stubEventVisualization();
+        final String eventDateDimension = EVENT_DATE.getDimension();
+        final Attribute column = COLUMN;
+        final SimpleDimensionHandler handler = new SimpleDimensionHandler( aEventVisualization );
+
+        // When
+        final DimensionalObject dimensionalObject = handler.getDimensionalObject( eventDateDimension, column );
+
+        // Then
+        assertThat( dimensionalObject.getUid(), is( equalTo( eventDateDimension ) ) );
+        assertThat( dimensionalObject.getDimensionType(), is( equalTo( PERIOD ) ) );
+        assertThat( dimensionalObject.getItems(), hasSize( 2 ) );
+    }
+
+    @Test
+    void testGetDimensionalObjectWhenDimensionIsNotValid()
+    {
+        // Given
+        final EventVisualization aEventVisualization = stubEventVisualization();
+        final String invalidDimension = "invalidDimension";
+        final Attribute column = COLUMN;
+        final SimpleDimensionHandler handler = new SimpleDimensionHandler( aEventVisualization );
+
+        // When throws
+        final NoSuchElementException thrown = assertThrows( NoSuchElementException.class,
+            () -> handler.getDimensionalObject( invalidDimension, column ) );
+
+        // Then
+        assertThat( thrown.getMessage(), containsString( "Invalid dimension: " + invalidDimension ) );
+    }
+
+    @Test
+    void testAssociateDimensions()
+    {
+        // Given
+        final String eventDateDimension = EVENT_DATE.getDimension();
+        final EventVisualization aEventVisualization = stubEventVisualization();
+        aEventVisualization.getSimpleDimensions().clear();
+        aEventVisualization.getColumns().addAll( asList( new BaseDimensionalObject( eventDateDimension ) ) );
```

Comment:
Can you use List.of(), instead of  asList? ( asList has only one argument)

Reply:
Done